resolver/resolver.cc

```diff
@@ -552,46 +552,61 @@ class ResolveConstantsWalk {
+                                send->fun.data(gs)->show(gs));
+                }
+                return;
             }
-            return;
-        }
-        if (id->symbol.data(gs)->isClassOrModuleClass()) {
-            if (auto e =
-                    gs.beginError(core::Loc(todo.file, send->loc), core::errors::Resolver::InvalidMixinDeclaration)) {
-                e.setHeader("`{}` is a class, not a module; Only modules may be mixins", id->symbol.data(gs)->show(gs));
+            if (id->symbol.data(gs)->isClassOrModuleClass()) {
+                if (auto e =
+                        gs.beginError(core::Loc(todo.file, id->loc), core::errors::Resolver::InvalidMixinDeclaration)) {
+                    e.setHeader("`{}` is a class, not a module; Only modules may be mixins",
+                                id->symbol.data(gs)->show(gs));
+                }
+                encounteredError = true;
             }
-            return;
-        }
-        if (id->symbol == owner) {
-            if (auto e =
-                    gs.beginError(core::Loc(todo.file, send->loc), core::errors::Resolver::InvalidMixinDeclaration)) {
-                e.setHeader("Must not pass your self to `{}`", send->fun.data(gs)->show(gs));
+            if (id->symbol == owner) {
+                if (auto e =
+                        gs.beginError(core::Loc(todo.file, id->loc), core::errors::Resolver::InvalidMixinDeclaration)) {
+                    e.setHeader("Must not pass your self to `{}`", send->fun.data(gs)->show(gs));
+                }
+                encounteredError = true;
             }
-            return;
-        }
-        auto existing = owner.data(gs)->findMember(gs, core::Names::classMethods());
-        if (existing.exists() && existing != id->symbol) {
-            if (auto e =
-                    gs.beginError(core::Loc(todo.file, send->loc), core::errors::Resolver::InvalidMixinDeclaration)) {
-                e.setHeader("Redeclaring `{}` from module `{}` to module `{}`", send->fun.data(gs)->show(gs),
-                            existing.data(gs)->show(gs), id->symbol.data(gs)->show(gs));
+            if (encounteredError) {
+                continue;
             }
-            return;
+
+            // Get the fake property holding the mixes
+            auto mixData = id->symbol.data(gs)->findMember(gs, core::Names::mixedInClassMethods());
+            auto loc = core::Loc(owner.data(gs)->loc().file(), send->loc);
+            if (!mixData.exists()) {
+                // We never stored a mixin in this symbol
+                // Create a the fake property that will hold the mixed in modules
+                mixData = gs.enterMethodSymbol(loc, owner, core::Names::mixedInClassMethods());
+            }
+
+            // Add an argument to the fake property to store the referenced module
+            auto argName = gs.enterNameUTF8(fmt::format("arg{}", argumentNumber)); // name has to be distinct
```

Comment:
There's a bug here: multiple calls to `mixes_in_class_methods` will reset `argumentNumber` to zero each time they are processed, so you'll end up with names that collide. You could just use the size of the arguments vector minus one as the number instead.

Reply:
I used `dataAllowingNone` to access the arguments vector but tests are failing due to `!symbolTableFrozen` [check](https://github.com/sorbet/sorbet/blob/master/core/GlobalState.cc#L1203) after that change so I'm looking for alternatives.

Reply:
I was able to confirm it only happens during tests and not during `dbg` builds of Sorbet. While running Sorbet directly this code is executed only once with `owner : module ::Mixin ()`.

In a test run however, in addition to that we have a second iteration with `owner : module ::Mixin < ::Sorbet::Private::Static::ImplicitModuleSuperclass () `. Also interestingly its `Loc` information within the test file points to a non existing line number.

During this run we call `enterMethodArgumentSymbol` which fails the `!symbolTableFrozen` check. Still investigating.